pkgs/development/python-modules/taichi/default.nix

```diff
@@ -0,0 +1,250 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  autoPatchelfHook,
+  autoAddDriverRunpath,
+  llvmPackages_15,
+  cmake,
+  ninja,
+  setuptools,
+  wheel,
+  pybind11,
+  scikit-build,
+  git,
+  pythonImportsCheckHook,
+  pythonCatchConflictsHook,
+
+  colorama,
+  dill,
+  matplotlib,
+  numpy,
+  rich,
+
+  cudaPackages,
+  elfutils,
+  libbfd,
+  libdwarf,
+  libffi,
+  libGL,
+  libX11,
+  libXrandr,
+  libXinerama,
+  libXcursor,
+  libXi,
+  libxml2,
+  vulkan-headers,
+  vulkan-loader,
+
+  gtest,
+  pytest,
+
+  config,
+  cudaSupport ? config.cudaSupport,
+  rocmSupport ? config.rocmSupport,
+  stdenv,
+  enableMetal ? stdenv.isDarwin,
+  enableOpenGL ? stdenv.isLinux,
+  enableVulkan ? stdenv.isLinux,
+}:
+
+let
+  pname = "taichi";
+  version = "1.7.3";
+  src = fetchFromGitHub {
+    owner = "taichi-dev";
+    repo = pname;
+    tag = "v${version}";
+    hash = "sha256-QWtxPgR8RGnIEpbGO18Y8TFNhbKd4pOi9VLEkK2nhfk=";
+    fetchSubmodules = true;
+    # leaveDotGit = true;
+  };
+
+  llvmPackages = llvmPackages_15;
+  libllvm = llvmPackages.libllvm;
+  stdenv = llvmPackages.stdenv;
+
+  platformLibs =
+    lib.optionals cudaSupport [ cudaPackages.cudatoolkit ]
+    ++ lib.optionals enableMetal [ ]
+    ++ lib.optionals rocmSupport [ ]
+    ++ lib.optionals enableVulkan [
+      vulkan-loader
+      vulkan-headers
+    ]
+    ++ lib.optionals enableOpenGL [ libGL ];
+
+  cmakeFlag =
+    flag: cflag:
+    "-D" + cflag + "=" + (if (lib.typeOf flag == "bool") then (if flag then "ON" else "OFF") else flag);
+
+  cmakeFlags = lib.attrsets.mapAttrsToList (name: value: (cmakeFlag value name)) {
```

Comment:
isn't this cmakeFlag args inverted?

Reply:
I took that function from a different package, since it made it a bit easier to specify things. I guess they flipped the `name` and `value`, and `flag` and `cflag` variables. I changed it to be the more logical order now.